File: projects/fizzbuzz/python/wizmariefa/fizzbuzzbazz.py
def fizzbuzz(n):
    string = ''
    for num in range(1, n+1):
        if num % 3 == 0:
            string += 'Fizz'
        if num % 5 == 0:
            string += 'Buzz'
        if num % 7 == 0:
            string += 'Bazz'
        if num % 3 != 0 and num % 5 != 0 and num % 7 != 0:
            string += str(num)
        
        string += "\n"
    return string

def valid_input(n):
    if n > 1:
        return n
    else:
        return -1

if __name__ == '__main__': # pragma: no cover
    n = -1
    while n == -1:
        n = int(input("Please enter your fizzbuzz number > "))
        n = valid_input(n)
    print(fizzbuzz(n))

File: projects/fizzbuzz/C/TimParrish/FizzBuzzBazz.c
#include <string.h>
#include <stdio.h>

char result[20];

void processUserInput(int passedNum){
    char convertedInt[20];
    int i = 0;
    for (i=1; i <= passedNum; i++){
        memset(result,0,strlen(result));
        memset(convertedInt,0,strlen(convertedInt));

        if (i%3==0 || i%5==0 || i%7==0){
            if (i%3 == 0){
                strncat(result, "Fizz", 4);
            }
            if (i%5 == 0){
                strncat(result, "Buzz", 4);
            }
            if (i%7==0){
                strncat(result, "Bazz", 4);
            }
        }
        else{
            sprintf(convertedInt, "%d", i);
            strncat(result, convertedInt, 20);
        }
#ifndef TESTBUILD
    printf("%s\n", result);
#endif
    }
    return;
}


#ifndef TESTBUILD
int main(){
    int userInput = 0;
    int valid = 0;

    while (valid == 0){
        printf ("Please enter a FizzBuzz number: ");
        scanf ("%d", &userInput);
        printf("w/ n = %d\n\n", userInput);
        if (userInput < 1 ){
            printf("Please enter an integer greater than or equal to 1\n");
        }
        else{
            valid = 1;
        }
    }
    processUserInput(userInput);
return 0;
}
#endif
